### ISSUE TITLE
WorkspaceEdit.changes key should be TextDocumentIdentifier

### ISSUE THREAD

**Issue by vladdu** [Microsoft/language-server-protocol#48]
Unless I'm misunderstanding, weren't simple URIs replaced with TextDocumentIdentifiers elsewhere? Then I'd say that it should be dine even here.

**Comment by dbaeumer** [comment]
The problem is that we simply reused JS Maps (Objects) here. The correct approach here is to return and array with object literals of the form:
```json
{
   document: TextDocumentIdentifier;
   changes: TextEdit[];
}
```

Since this is breaking it has to wait until 3.0.

**Comment by dbaeumer** [comment]
This got addressed for 3.0. You can have a workspace edit with versioned text document identifiers.
